src/components/engagement/engagement.resolver.ts

```diff
@@ -44,7 +44,10 @@ export class EngagementResolver {
     @Args() { id, changeset }: ChangesetIds,
     @AnonSession() session: Session
   ): Promise<Engagement> {
-    const engagement = await this.service.readOne(id, session, changeset);
+    const engagement = await this.service.readOne(id, session, {
+      type: 'changeset',
+      value: changeset,
+    });
```

Comment:
This needs to use active if changeset is not given. Maybe there's someway we could convert `ChangesetIds` to an `ObjectView`